modules/storages/app/services/storages/group_folder_properties_sync_service.rb

```diff
@@ -28,266 +28,268 @@
-  def admins_project_folder_permissions
-    @admin_nextcloud_usernames.each_with_object(
-      "#{@nextcloud_system_user}": ALL_PERMISSIONS
-    ) do |admin_nextcloud_username, hash_map|
-      hash_map[admin_nextcloud_username.to_sym] = ALL_PERMISSIONS
+      if project_storage.project.public? && ProjectRole.non_member.permissions.intersect?(PERMISSIONS_KEYS)
+        project_tokens
+      else
+        project_tokens.where(user: project_storage.project.users)
+      end
     end
-  end
 
-  def members_project_folder_permissions(project:)
-    tokens_query(project:).each_with_object({}) do |token, permissions|
-      nextcloud_username = token.origin_user_id
-      permissions[nextcloud_username.to_sym] = calculate_permissions(user: token.user, project:)
-      @nextcloud_usernames_used_in_openproject << nextcloud_username
+    def hide_inactive_folders(remote_folders)
+      project_folder_ids = active_project_storages_scope.pluck(:project_folder_id).compact
+      remote_folders.except("#{@storage.group_folder}/").each do |(path, attrs)|
+        next if project_folder_ids.include?(attrs['fileid'])
+
+        command_params = {
+          path:,
+          permissions: {
+            users: { "#{@storage.username}": ALL_PERMISSIONS },
+            groups: { "#{@storage.group}": NO_PERMISSIONS }
+          }
+        }
+
+        Peripherals::Registry
+          .resolve("commands.nextcloud.set_permissions")
+          .call(storage: @storage, **command_params)
+          .result_or do |error|
+          OpenProject.logger.warn({ command: 'nextcloud.set_permissions',
+                                    folder: path,
+                                    message: error.log_message,
+                                    data: { status: error.data.code, body: error.data.body } }.to_json)
+        end
+      end
     end
-  end
 
-  def tokens_query(project:)
-    tokens_query = OAuthClientToken
-                     .where(oauth_client: @storage.oauth_client)
-                     .where.not(id: @admin_tokens_query)
-                     .includes(:user)
-                     .order(:id)
-    # The user scope is required in all cases except one:
-    #   when the project is public and non member has at least one storage permission
-    #   then all non memebers should have access to the project folder
-    if !(project.public? && ProjectRole.non_member.permissions.intersect?(PERMISSIONS_KEYS))
-      tokens_query = tokens_query.where(users: project.users)
+    def ensure_folders_exist(remote_folders)
+      id_folder_map = remote_folders.to_h { |folder, properties| [properties['fileid'], folder] }
+
+      active_project_storages_scope.map do |project_storage|
+        next create_folder(project_storage) unless id_folder_map.key?(project_storage.project_folder_id)
+
+        current_path = id_folder_map[project_storage.project_folder_id]
+        if current_path == project_storage.project_folder_path
+          project_storage.project_folder_id
```

Comment:
It seems we do not need `project_folder_id` value up on the stack and it has been used as a boolean. Let's return a boolean explicitly, maybe?

Reply:
I'm sorry, but I'm kinda slow today. What do you mean in this case? Could you express this in code?

Reply:
Nevermind. After having a closer look I see that what I suggested does not make sense. Current implementation is fine :+1: